processors/src/main/scala/org/clulab/processors/bionlp/ner/KBGenerator.scala

```diff
@@ -98,59 +95,16 @@ object KBGenerator {
     entries.toList
   }
 
-  def processKBFiles():Map[String, Seq[String]] = {
-
-//    val entries = loadConfig(configFile)
-    val entries = loadFromConf()
-    logger.info(s"Will convert a total of ${entries.size} KBs:")
-
-    (for(entry <- entries) yield {
-      val processedLines = convertKB(entry)
-      entry.kbName -> processedLines
-    }).groupBy(_._1).mapValues(_.flatMap(_._2))
-  }
-
-  def convertKB(entry:KBEntry): Seq[String] = {
-    logger.info(s"Loading ${entry.kbName}...")
-    val inputPath = entry.path//inputDir + File.separator + entry.kbName + ".tsv"
-    val b =
-      if(new File(inputPath).exists())
-        new BufferedReader(new InputStreamReader(new FileInputStream(inputPath)))
-      else
-        new BufferedReader(new InputStreamReader(new GZIPInputStream(new FileInputStream(inputPath + ".gz"))))
-
-    val tokenizer: Tokenizer = (new BioNLPProcessor).tokenizer
-
-    var done = false
-    var lineCount = 0
-    val outputLines = new ArrayBuffer[String]()
-    while(! done) {
-      val line = b.readLine()
-      if(line == null) {
-        done = true
-      } else {
-        val trimmedLine = line.trim
-        if(trimmedLine.nonEmpty && ! trimmedLine.startsWith("#")) { // skip comments
-          val kbTokens = line.split("\t")
-          if(containsValidSpecies(entry, kbTokens)) { // this is a protein from a species we want
-            lineCount += 1
-            val ne = kbTokens(0) // we enforce that the first token is the actual NE to be considered
-            val tokens = tokenizeResourceLine(ne, tokenizer) // tokenize using BioNLPProcessor
-            outputLines += tokens.mkString(" ")
-          }
-        }
-      }
-    }
-    b.close()
-
-
-    val uniqLines = outputLines
-      .filter(_.nonEmpty)
-      .sorted
-      .distinct
+  def processKBFiles(caseInsensitiveMatching: Boolean): Seq[StandardKbSource] = {
+    val kbEntries = loadFromConf()
+    // This should only be performed on small collections.
+    // It is important to keep both the labels and the KbEntries in the same order.
+    val actualLabels = kbEntries.map(_.neLabel).distinct
+    val expectedLabels = Seq("Gene_or_gene_product", "Family", "Cellular_component", "Simple_chemical", "Site", "BioProcess", "Disease", "Species", "CellLine", "TissueType", "CellType", "Organ")
+    val kbEntriesSeq = expectedLabels.map { label => kbEntries.filter(_.neLabel == label) }
```

Comment:
This line and line 103 should go away for two reasons:
1. `loadFromConf` will guarantee the entries come in the order specified by the user in the configuration
2. A new KB file with a previously unseen label can be added by a user

I am sorry for leading you to port the hard-coded iteration order

Reply:
I understand and expected 103, but what is "This line"?  I'm missing the antecedent.

Reply:
Yes, thanks, that line would use the actualLabels again.